### PR TITLE
Feat/addditional x

### PR DESCRIPTION
**Pull request by meowsbits**
Rel #16

### CODE REVIEW

File: src/index.test.ts [hunk 1]
```diff
@@ -51,6 +51,72 @@ describe("traverse", () => {
...
+      traverse(testSchema, mockMutation);
+
+      expect(mockMutation).toHaveBeenCalledWith(testSchema.additionalProperties);
+      expect(mockMutation).toHaveBeenCalledWith(testSchema);
+      expect(mockMutation).toHaveBeenCalledTimes(2);
+    });
+
+    it("traverses additionalProperties as schema", () => {
+      const testSchema: any = {
+        additionalProperties: {
```
Comment (BelfordZ) [inline src/index.test.ts:69]: While i know that the type is implied when there is a properties field, I condemn it as bad practice, and would propose that we make type required in our own json schema meta-schema.
```suggestion
        additionalProperties: {
          type: "object",
```

File: src/index.test.ts [hunk 2]
```diff
@@ -51,6 +51,72 @@ describe("traverse", () => {
...
+          properties: {
```
Comment (BelfordZ) [inline src/index.test.ts:70]: I'd like to make some more test cases, but this is a good start

**Comment by BelfordZ** [comment]
@meowsbits tests are failing, and its not the tests fault. The issue is that we currently mis-handle type: [], so using a boolean as a schema ATM isnt really working 100%. We can leave this test failing until that support is added, or comment out the test until I have the fix in for type: [], up to you.

**Comment by meowsbits** [comment]
🐿

**Comment by BelfordZ** [comment]
all fixed up. just needs rebase now

**Comment by meowsbits** [comment]
@BelfordZ rebased with fixes for minor conflicts in src/index.ts all preferring master.

**Comment by BelfordZ** [comment]
Pushed the changes that we worked on while pairing Friday June 12th

**Pull request merged**